dockstore-client/src/main/java/io/dockstore/client/cli/nested/ApiClientExtended.java

```diff
@@ -109,4 +138,192 @@ public void createBodyPart(MultiPart multiPart, String key, Object formObject) {
         return entity;
     }
 
+    /**
+     * Invoke API by sending HTTP request with the given options.
+     *
+     * @param <T> Type
+     * @param path The sub-path of the HTTP URL
+     * @param method The request method, one of "GET", "POST", "PUT", "HEAD" and "DELETE"
+     * @param queryParams The query parameters
+     * @param body The request body object
+     * @param headerParams The header parameters
+     * @param formParams The form parameters
+     * @param accept The request's Accept header
+     * @param contentType The request's Content-Type header
+     * @param authNames The authentications to apply
+     * @param returnType The return type into which to deserialize the response
+     * @return The response body in type of string
+     * @throws ApiException API exception
+     */
+    @Override
+    @SuppressWarnings("checkstyle:ParameterNumber")
+    public <T> T invokeAPI(String path, String method, List<Pair> queryParams, Object body, Map<String, String> headerParams, Map<String, Object> formParams, String accept, String contentType, String[] authNames, GenericType<T> returnType) throws ApiException {
+        updateParamsForAuth(authNames, queryParams, headerParams);
+
+        // Not using `.target(this.basePath).path(path)` below,
+        // to support (constant) query string in `path`, e.g. "/posts?draft=1"
+        WebTarget target = httpClient.target(this.basePath + path);
+
+        if (queryParams != null) {
+            for (Pair queryParam : queryParams) {
+                if (queryParam.getValue() != null) {
+                    target = target.queryParam(queryParam.getName(), queryParam.getValue());
+                }
+            }
+        }
+
+        final boolean requiresAwsHeaders = this.wesRequestData.getCredentialType() == WesRequestData.CredentialType.AWS_PERMANENT_CREDENTIALS;
```

Comment:
Turned this into a function. Especially helpful for when we add temporary credentials, as otherwise you'd need to do multiple comparisons (`is aws_permanent` or `is aws_temporary`) or invert it (`is bearer_token`).

https://github.com/dockstore/dockstore-cli/pull/85/commits/a7558e9eee0a3b943b77fe46b474d5bef411d397